packages/@uppy/aws-s3/src/index.ts

```diff
@@ -486,15 +536,17 @@ export default class AwsS3Multipart<
     throwIfAborted(options?.signal)
 
     if (this.#cachedTemporaryCredentials == null) {
+      const { getTemporarySecurityCredentials } = this.opts
       // We do not await it just yet, so concurrent calls do not try to override it:
-      if (this.opts.getTemporarySecurityCredentials === true) {
+      if (getTemporarySecurityCredentials === true) {
         this.#assertHost('getTemporarySecurityCredentials')
         this.#cachedTemporaryCredentials = this.#client
           .get<AwsS3STSResponse>('s3/sts', options)
           .then(assertServerError)
       } else {
         this.#cachedTemporaryCredentials =
-          this.opts.getTemporarySecurityCredentials(options)
+          // @ts-expect-error It's either a function or `false`, which should error.
+          getTemporarySecurityCredentials(options)
```

Comment:
```suggestion
          if (typeof getTemporarySecurityCredentials !== 'function') throw new Error('Expected getTemporarySecurityCredentials to be "true" or a function')
          getTemporarySecurityCredentials(options)
```
better to check for it. `@ts-expect-error` will blanket disable all errors. e.g. a wrong `options` type.

Reply:
```
This expression is not callable.
  Type 'never' has no call signatures.
```

Reply:
is `getTemporarySecurityCredentials` `never`? is it possible to fix the types of `getTemporarySecurityCredentials` so that it's not `never`?